### PR TITLE
Added 'DryRun' switch for pre-check in Storage account control

### PR DESCRIPTION
**Pull request by Aditi-sha**
Added -DryRun switch as pre-check, if user wants to remediate control at Storage account level.
Also added DisplayName in NonAD control script.

### CODE REVIEW

File: Scripts/RemediationScripts/Remediate-AnonymousAccessOnContainers.ps1 [hunk 1]
```diff
@@ -27,17 +29,19 @@ ControlId:
...
-        2. Run below command to remove anonymous access from given storage account(s) of subscription
-
-        Remove-AnonymousAccessOnContainers -SubscriptionId '<Sub_Id>' -RemediationType '<DisableAnonymousAccessOnContainers>, <DisableAllowBlobPublicAccessOnStorage>'  -Path '<Json file path containing storage account(s) detail>' [-ExcludeResourceGroupNames <Comma separated resource group name(s) to be excluded from remediation>] [-ExcludeResourceNames <Comma separated resource name(s) to be excluded from remediation>]
-
-    Note: 
-        To rollback changes made by remediation script, execute below command
-        Set-AnonymousAccessOnContainers -SubscriptionId '<Sub_Id>' -RollBackType '<EnableAnonymousAccessOnContainers>, <EnableAllowBlobPublicAccessOnStorage>'  -Path '<Json file path containing Remediated log>'   
+        Note:
+            i. [Recommended] Use -DryRun parameter to get details of Storage accounts in CSV for pre-check.
+            ii. DryRun check is only available, if you are remediating conttrol at Storage account level (not at container level).
```
Comment (TarunKrShukla) [inline Scripts/RemediationScripts/Remediate-AnonymousAccessOnContainers.ps1:40]: typo conttrol  --> control

Comment (Aditi-sha) [reply Scripts/RemediationScripts/Remediate-AnonymousAccessOnContainers.ps1]: Corrected.

File: Scripts/RemediationScripts/Remediate-AnonymousAccessOnContainers.ps1 [hunk 2]
```diff
@@ -195,7 +205,7 @@ function Remove-AnonymousAccessOnContainers
         break;
     }
     
-    Write-Host "Successfully validated" -ForegroundColor $([Constants]::MessageType.Update)
```
Comment (TarunKrShukla) [inline Scripts/RemediationScripts/Remediate-AnonymousAccessOnContainers.ps1:198]: We are just checking for User type not the RBAC role, is this pattern followed in all scripts.

File: Scripts/RemediationScripts/Remediate-AnonymousAccessOnContainers.ps1 [hunk 3]
```diff
@@ -316,10 +326,20 @@ function Remove-AnonymousAccessOnContainers
...
+                        Write-Host "Path: $($folderPath)DisabledAllowBlobPublicAccess.csv"
+                        return;
+                    }
+                    else
+                    {
+                        Write-Host "Backing up config of storage account(s) details. Please do not delete this file. Without this file you won't be able to rollback any changes done through remediation script." -ForegroundColor $([Constants]::MessageType.Info)
+                        $stgWithEnableAllowBlobPublicAccess | ConvertTo-json | out-file "$($folderpath)\DisabledAllowBlobPublicAccess.json"  
+                        Write-Host "Path: $($folderpath)\DisabledAllowBlobPublicAccess.json"
+                    }
+
```
Comment (TarunKrShukla) [inline Scripts/RemediationScripts/Remediate-AnonymousAccessOnContainers.ps1:342]: ResourceResolver check if this is loaded or not?

Comment (TarunKrShukla) [reply Scripts/RemediationScripts/Remediate-AnonymousAccessOnContainers.ps1:342]: At line # 306 Use Get-Member instead.

File: Scripts/RemediationScripts/Remediate-AnonymousAccessOnContainers.ps1 (hunk 2 above)
Comment (Aditi-sha) [reply Scripts/RemediationScripts/Remediate-AnonymousAccessOnContainers.ps1:198]: Added role check for Owner, Contributor and UAA.

File: Scripts/RemediationScripts/Remediate-AnonymousAccessOnContainers.ps1 (hunk 3 above)
Comment (Aditi-sha) [reply Scripts/RemediationScripts/Remediate-AnonymousAccessOnContainers.ps1:365]: Added try-catch for ResourceResolver, if module not loaded.

Comment (Aditi-sha) [reply Scripts/RemediationScripts/Remediate-AnonymousAccessOnContainers.ps1:364]: Updated condition with Get-Member.

**Review by TarunKrShukla — APPROVED** [review]

**Review by Aboli-msft — APPROVED** [review]

**Pull request merged**